lib/core/cardano-wallet-core.cabal

```diff
@@ -173,6 +189,8 @@ library
       Crypto.Hash.Utils
       Data.Binary.Get.Safe
       Data.Function.Utils
+      Data.Map.NonEmpty.Strict
+      Data.Map.NonEmpty.Strict.Internal
```

Comment:
@KtorZ wrote:
> Have you considered existing hackage / stackage alternatives? There's https://hackage.haskell.org/package/nonempty-containers although it is lazy-only.

I did indeed look through Hackage for alternatives. There are several non-empty maps, but sadly they all seem to be implemented in terms of lazy maps.

> Beside, may I consider putting this in a separate package. This is decoupled enough from the wallet to live out of it.

We could do that if you prefer. How about `strict-non-empty-containers`? (Unfortunately, `non-empty-containers` is already used.)

Having said that, we don't really have a need for it outside of `cardano-wallet-core`, so we probably wouldn't gain too much from moving it (yet).

Reply:
I am not saying we should **publish** a new package, but simply organize our internal code differently. Since this is clearly not wallet-specific, there's no reason to keep it in `cardano-wallet-core`; it'll prevent the temptation from adding wallet-specific things to it.

Reply:
> I am not saying we should publish a new package, but simply organize our internal code differently.

Oh definitely! I didn't think that was your suggestion. Sorry if my comment gave that impression. 😄 

I was just trying to come up with an internal package name. My proposal would be something like `strict-non-empty-containers`.

Reply:
@KtorZ I've moved this type (and its tests) to a new internal package `strict-non-empty-containers`.